pkg/vault/contracts/authorizer/TimelockAuthorizer.sol

```diff
@@ -407,18 +428,21 @@ contract TimelockAuthorizer is IAuthorizer, IAuthentication, ReentrancyGuard {
     }
 
     /**
-     * @notice Returns true if `account` can perform action `actionId` in target `where`.
-     * @dev All authentications that require the authorizer adaptor must originate from the authorizer adaptor
-     * entrypoint: requests coming directly from the authorizer adaptor will be rejected.
+     * @notice Returns true if `account` can perform action `actionId` in target `where`. This will return false for
+     * actions that have a delay associated with them, even if `account` has permission over the action, since `account`
+     * cannot perform the action directly - it must instead schedule a future execution of it via `schedule`.
+     *
+     * @dev All authentications that require the AuthorizerAdaptor must originate from the AuthorizerAdaptorEntrypoint:
+     * requests coming directly from the AuthorizerAdaptor will be rejected.
      */
     function canPerform(
         bytes32 actionId,
         address account,
         address where
     ) public view override returns (bool) {
         if (msg.sender == address(_authorizerAdaptor)) {
-            // We special case the situation where the caller is the `AuthorizerAdaptor`, as it can be tricked into
-            // passing an incorrect `actionId` value, potentially resulting in escalation of privileges.
+            // We special case the situation where the caller is the `AuthorizerAdaptor`, as due to a bug it can be
```

Comment:
```suggestion
            // In the situation where the caller is the `AuthorizerAdaptor`, as due to a bug it can be
```

Reply:
This isn't wrong even as written, since "special case" is commonly used as a verb like that, but the rewrite is even clearer.